h2o-core/src/main/java/hex/RegexTokenizer.java

```diff
@@ -0,0 +1,98 @@
+package hex;
+
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+import water.parser.BufferedString;
+
+/**
+ * RegexTokenizer splits rows of a given Frame into delimited sequences of tokens using a regular expression.
+ * The output structure is suitable for use in the word2vec algorithm.
+ *
+ * <p>
+ * Example usage:
+ * <pre>{@code
+ * final RegexTokenizer tokenizer = new RegexTokenizer.Builder()
+ *   .setRegex("[,;]")
+ *   .setMinLength(2)
+ *   .setToLowercase(true)
+ *   .create();
+ * final Frame tokens = tokenizer.transform(inputFrame);
+ * }
+ * </pre>
+ */
+public class RegexTokenizer extends MRTask<RegexTokenizer> {
+  private final String _regex;
+  private final boolean _toLowercase;
+  private final int _minLength;
+
+  public RegexTokenizer(String regex) {
+    this(regex, false, 0);
+  }
+
+  private RegexTokenizer(String regex, boolean toLowercase, int minLength) {
+    _regex = regex;
+    _toLowercase = toLowercase;
+    _minLength = minLength;
+  }
+
+  @Override
+  public void map(Chunk[] cs, NewChunk nc) {
```

Comment:
It would be nice to use some class instead of ```Chunk[]``` in future. I needed a help from @jakubhava to understand that this describes one partition across multiple columns.

Reply:
Just clarification -> it describes one partition in a vector, but there is ( at least as I see it) unwritten invariant that says that for example, all the first chunks across different vectors have to have same size

Reply:
Yes, that is a good idea - having this is also necessary for true sparse support.